docs/index.asciidoc

```diff
@@ -84,6 +84,35 @@ like `myapp-%{sequence_id}`. Reusing ids will cause documents to be rewritten.
   * There is no default value
 
 The Engine name. Engine is your search engine created in App Search, an information repository that includes the indexed document records.
+Engine field also support `sprintf` format, so a valid name could be `engine-%{eng_level}`, in this case the engine
+value is resolved at runtime, event by event; dynamic resolution could lead to invalid engine names in which case
```

Comment:
Thanks for suggestion @robbavey, applied :-)

I've a question about:
> extend the example to check the valid engine names

if I understand correctly, you mean to do an HTTP call to the AppSearch service to validate the resolved engine name, this could double the calls to AppSearch, one for the check and one for effectively index the data. To avoid some calls the pipeline should provide some form of caching, keeping the already validated engine names. This could be done with `ruby` filter maybe. Is it what you mean?

Reply:
Actually, my initial thought for an example was something simpler - just a manual check that the engine name is in a list of engine names that the user would provide in the configuration. Not a great solution in the long term, depending on how many engines a user would typically include in a cluster, and how often that list changes, but it may be of some help